src/app/scripts/cac/pages/cac-pages-directions.js

```diff
@@ -23,17 +23,55 @@ CAC.Pages.Directions = (function ($, _, DirectionsList, Itinerary, Settings, Use
     }
 
     Directions.prototype.initialize = function () {
-        var params = Utils.getUrlParams();
-        if (!_.has(params, 'itineraryIndex')) {
-            // TODO: show this error in the UI
+
+        // Note: date/time does not get passed and so always defaults to departing now
+
+        var rawParams = location.search.slice(1).split('&');
+        var itineraryIndex = -1; // initialize to flag value for missing index parameter
+
+        // process parameters for consumption by OpenTripPlanner:
+        // strip out itineraryIndex, and convert singe waypoint parameter into
+        // multiple intermediatePlaces parameters.
+        var otpParams = _.map(rawParams, function(paramStr) {
```

Comment:
I also do not love all the `indexOf` footwork.  Here's a version that breaks up the initial param string further, into an array of `[paramName, value]`arrays, then defines rewriting rules and parses the array using them.

It's not necessarily shorter, and it's not devastatingly readable, but I think it's more direct and more flexible.

```
var rawParams = _.map(location.search.slice(1).split('&'), function(p) { 
    return p.split('=');
});

// Pull out the itinerary index and validate it
var itineraryIndex = _.remove(rawParams, function (p) { return p[0] === 'itineraryIndex';});
if (itineraryIndex.length !== 1) {
    console.error('Must specify itineraryIndex URL parameter');
    return;
} else {
    itineraryIndex = parseInt(itineraryIndex[0][1]);
    if (isNaN(itineraryIndex)) {
        console.error('itineraryIndex URL parameter must be an integer');
        return;
    }
}

// Rules for rewriting GoPhillyGo URL params to OTP query params.
// If 'values' is defined, parse out the value into multiple copies of the param,
// one per item in the array returned by calling 'values' with the URL parameter value.
// Any URL param not in the mapping is passed through unchanged.
var mapping = {
    origin: { paramName: 'fromPlace' },
    originText: { paramName: 'fromText' },
    destination: { paramName: 'toPlace' },
    destinationText: { paramName: 'toText' },
    waypoints: {
        paramName: 'intermediatePlaces',
        values: function (val) { 
            return _.map(decodeURIComponent(val).split(';'), encodeURIComponent);
        }
    },
};

// Rewrites an array of [URLparam, value] pairs into an array of [OTPparam, value] pairs,
// using the mapping above.
var otpParams = _(rawParams).map(function (param) {
    if (!mapping[param[0]]) {
        return [param];
    } else {
        var config = mapping[param[0]];
        if (!config.values) {
            return [[config.paramName, param[1]]];
        } else {
            return _.map(config.values(param[1]), function (val) {
                return [config.paramName, val];
            });
        }
    }
}).flatten().value();
// Join array into param string
otpParams = _(otpParams).map(function (item) { return item.join('='); }).join('&');
```